Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/presenter/ImageViewerPresenter.java

```diff
@@ -432,6 +627,29 @@ private void setTitle(PostImage postImage, int position) {
         return other;
     }
 
+    public boolean forceReload() {
+        PostImage currentImage = getCurrentPostImage();
+
+        if (fileCacheV2.isRunning(currentImage.imageUrl.toString())) {
+            showToast("Image is not yet downloaded");
+            return false;
+        }
+
+        if (!cacheHandler.deleteCacheFileByUrl(currentImage.imageUrl.toString())) {
+            showToast("Can't force reload because couldn't delete cached image");
+            return false;
+        }
+
+        callback.setImageMode(currentImage, LOWRES, false);
+        return true;
+    }
+
+    private enum SwipeDirection {
+        Default,
```

Comment:
Yeah, we need Default state that does nothing (or it probably should be renamed as NoDirection) because when we open an image by clicking it we don't want to preload any other image and that's why we need the third state - Default in this case.

Reply:
Right, but what I mean is that generally the user will start swiping forward, so the default state seems pointless.

Reply:
Well can't know for sure. The user may start swiping backwards too.

Reply:
And one more reason why I did it like that is because preloading spams the logs and it's hard to see logs for the image you clicked (the one you care about) and not the one that is preloading.

Reply:
Right, but understand that the _USER_ doesn't give a shit about logs. The user cares about if things are working as they expect it to, which in this case probably means they'd expect the next image to be preloading, which is generally forward. It's a small change that improves UX and doesn't hurt too much if the user doesn't start swiping in the forward direction to start.

Reply:
It hurts if the user doesn't use swipes at all, only clicking at images and then clicking back. In such case we will preloading an image for nothing (I don't really use swipes at all, I only ever used them to test this thing).

Maybe this can be made into a separate setting? Something like: "When clicking an image do you want to preload next/previous/both/none image?"

Reply:
I'd be alright with this being a setting, even if I think it's pointless. The default of the setting should be next/forward though, as most people use the image viewer in that manner.

Reply:
Alright.

Reply:
Done.